delta/ml/train.py

```diff
@@ -389,23 +396,29 @@ def train(model_fn, dataset : ImageryDataset, training_spec, resume_path=None):
                                      callbacks=callbacks, verbose=1)
 
         if config.mlflow.enabled():
-            model_path = os.path.join(mcb.temp_dir, 'final_model.h5')
-            print('\nFinished, saving model to %s.' % (mlflow.get_artifact_uri() + '/final_model.h5'))
+            model_path = os.path.join(mcb.temp_dir, 'final_model' + config.train.default_model_extension())
```

Comment:
I like how in the save function you checked if the extension was h5 and chose based on that. For the train command, an output filename can be given. Why don't instead of adding another field, we use whatever extension was given there? It could be an extra argument to the train function here, which in the train command we would extract from that filename if given. (just I expect the behavior of this option would be confusing)

Reply:
I saw this in subcommands/train.py:
```
if options.model is not None:
            save_model(model, options.model)
```
Is options.model always going to be available?  If so I can make this change.

Reply:
In subcommands/train.py it will always be available but may be None, so just choose the default for that case. You won't have access to options in delta/ml/train.py so I would add an argument to the train function there like save_format='h5'